### ISSUE TITLE
PGCallableStatement problem with 0.8.6 version

### ISSUE THREAD

**Issue by butlnor** [impossibl/pgjdbc-ng#525]
When executing something like:

 - select abc from abc(($1)::bigint)

with version 0.8.6, I get an error, while the same statement otherwise is normally working when executing like in dBeaver and similar tools, and it was properly working with pgjdbc-ng versions 0.5 and 0.7.1.

The error reported is "NO_RESULT_SET_AVAILABLE".

**Comment by butlnor** [comment]
This is a minimal source code that exposes the problem:

```
PGDataSource ds = new PGDataSource();
// ds.set.... user, password, etc.
try (Connection conn = ds.getConnection()) {
	try (CallableStatement cs = conn.prepareCall("SELECT abc FROM abc((?)::bigint)")) {
		cs.setInt(1, 1);
		try (ResultSet rs = cs.executeQuery()) {
			// does not come here, the exception is thrown before
		}
	}
}
```

**Comment by butlnor** [comment]
The problem is, that inside PGCallableStatement.execute() method, the return value of "super.execute()" is not used anymore, like it was in previous versions, and that "return false" is hardcoded, instead of using the result of super.execute(). 

Please fix this back, this is urgent and critical to our project, which now cannot be deployed with the new PGJDBC-NG version.

**Comment by kdubb** [comment]
@butlnor Why are you using `CallableStatement` and not `PreparedStatement` (or `Statement`)?

**Comment by butlnor** [comment]
> 
> 
> @butlnor Why are you using `CallableStatement` and not `PreparedStatement` (or `Statement`)?

Because I didn't write that source code initially, I only inherited it. And because CallableStatement, regarding to the specifications, should work, and it was working in the previous versions of pgjdbc-ng. Also, the solution I mentioned, to return the return value from super.execute(), should be working.

However, since this was a critical issue for our project, I couldn't wait for your fix, so I did replace the CallableStatement with PreparedStatement. I though it was a temporary solution, but you are saying this should be a permanent one?
